acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+"""Testing spaces and subnets settings for app deployment to lxd container.
+   This is an validation of Bug 1685782:
+   https://bugs.launchpad.net/juju/+bug/1685782
+   The test procedures are based on comment #29.
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+import json
+import logging
+import argparse
+
+from jujucharm import local_charm_path
+from deploy_stack import BootstrapManager
+
+from utility import (
+    JujuAssertionError,
+    add_basic_testing_arguments,
+    configure_logging,
+    )
+
+__metaclass__ = type
+log = logging.getLogger("assess_lxd_space")
+
+
+def assert_initial_spaces(client):
+    """Initial spaces status after bootstrap should show as:
```

Comment:
At this stage this test is only for AWS because the main purpose is to validate the Bug 1685782 instead of thoroughly test space/subnets. I myself have no knowledge in MAAS space but as you mentioned, if the behaviour is different we then need to validate them differently - if such validation is deemed necessary. Actually in this case the assert_initial_spaces() can be removed because the purpose is to test in that documented scenario, if there is error after the second deployment, which is hardly relevant to the initial space status.  I put a switcher in assess_lxd_space() in case we need to handle other environment in future.